src/adagrams.js

```diff
@@ -1,15 +1,211 @@
+};
+
+// helper function for highestScoreFrom
+// to create an object of words and corresponding scores
+const createWordScoresObj = (wordsArr) => {
+  const wordScores = {};
+
+  for (let word of wordsArr) {
+    let score = scoreWord(word);
+    wordScores[word] = score;
+  }
+
+  return wordScores;
+};
+
+// helper function for highestScoreFrom to determine the high score
+const findHighScore = (wordScoresObj) => {
+  let highScore = 0;
+
+  for (let word in wordScoresObj) {
+    if (wordScoresObj[word] > highScore) {
+        highScore = wordScoresObj[word];
+    }
+  }
+
+  return highScore;
+};
+
+// helper function for highestScoreFrom
+// to create an array of high score words
+const createHighScoreWordsArray = (wordScoresObj, highScore) => {
+  let highScoreWords = [];
+
+  for (let word in wordScoresObj) {
+    if (wordScoresObj[word] === highScore) {
+      highScoreWords.push(word);
+    }
+  }
+
+  return highScoreWords;
+};
+
+
+// an array prototype to find the shortest string in an array and return it
+Array.prototype.findShortestString = function() {
+  if (this.length === 0) {
+    return null; // Return null if the array is empty
+  }
+
+  let shortestString = this[0];
+  for (let i = 1; i < this.length; i++) {
+    const currentString = this[i];
+    if (currentString.length < shortestString.length) {
+      shortestString = currentString;
+    }
+  }
+
+  return shortestString;
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  const maxLettersInHand = 10;
+
+  const letterFrequency = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+  };
+
+  // Create a letter "bowl" in an array to emulate a bowl of letters to choose from
+  // based on likelihood from letterFrequency
+  const letterBowl = createLetterBowl(letterFrequency);
+  
+  // build the player's hand using the count prototype
+  const hand = [];
+
+  let num = 0;
+  while (num < maxLettersInHand) {
+    // choose a random number from 0-97
+    let i = Math.floor(Math.random() * letterBowl.length)
+    let letter = letterBowl[i]
+    if (hand.count(letter) < letterFrequency[letter]) {
+      hand.push(letter);
+      num++;
+    }
+  }
+  
+  return hand
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+
+  // make input case insensitive and turn it into an array
+  // so that we can use the custom count prototype on it
+  input = input.toUpperCase().split('');
+
+  // to ensure each letter is in lettersInHand and does not occur too frequently:
+  for (let letter of input) {
+    if (!(lettersInHand.indexOf(letter) !== -1) || (input.count(letter) > lettersInHand.count(letter))) {
```

Comment:
I corrected this double negative conditional but cannot figure out how to update the pull request, just fyi 🥲